providers/universal-provider/src/utils/caip25.ts

```diff
@@ -54,7 +54,15 @@ export const extractCapabilitiesFromSession = (
   // get all capabilities from sessionProperties as they apply to all chains/addresses
   const globalCapabilities = getCapabilitiesFromObject(sessionProperties);
 
-  for (const chain of chainIds) {
+  const namespaceChainIds =
+    namespaces[EIP155_PREFIX]?.chains
+      ?.map((chain) => decimalToHex(chain.split(":")[1]))
+      .filter(Boolean) ?? [];
```

Comment:
```suggestion
    const namespaceChainIds =
    namespaces[EIP155_PREFIX]?.accounts
      .filter((account) => account.includes(`:${address}`))
      ?.map((account) => decimalToHex(parseChainId(account).reference))
      .filter(Boolean) ?? [];
```
Check the accounts because we don't want to get all chains, just the chains this address was approved with

tip: you can import `parseChainId` from `"@walletconnect/utils"`